digdag-core/src/main/java/io/digdag/core/agent/OperatorManager.java

```diff
@@ -110,24 +118,27 @@ public void run(TaskRequest request)
     {
         long taskId = request.getTaskId();
 
-        // set task name to thread name so that logger shows it
-        try (SetThreadName threadName = new SetThreadName(request.getTaskName())) {
-            try (TaskLogger taskLogger = callback.newTaskLogger(request)) {
-                TaskContextLogging.enter(LogLevel.DEBUG, taskLogger);
-                try {
-                    runningTaskMap.put(taskId, request);
+        tm.begin(() -> {
```

Comment:
I think that starting transaction here is not good idea because the task runs in a transaction. All locks acquired in this transaction won't be released until the user task finishes.
`TaskCallbackApi` (and `SecretProvider`) are the only interface for agents to access database. Therefore, you can put `tm.begin` in TaskCallbackApi` to make all DB access from tasks transactional.